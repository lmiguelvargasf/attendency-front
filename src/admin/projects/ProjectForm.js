import React from 'react'
import { Form, Input, Button, DatePicker } from 'antd'
import PropTypes from 'prop-types'

const layout = {
  labelCol: { span: 8 },
  wrapperCol: { span: 8 }
}
const tailLayout = {
  wrapperCol: { offset: 8, span: 8 }
}

const ProjectForm = ({ project, onFinish, onFinishFailed }) => {
  return (
    <Form
      data-testid='create-project-form'
      {...layout}
      name='project'
      initialValues={project}
      onFinish={onFinish}
      onFinishFailed={onFinishFailed}
    >
      <Form.Item
        label='Title'
        name='title'
        rules={[{ required: true, message: 'Please input the title!' }]}
      >
        <Input data-testid='project-title-input' />
      </Form.Item>

      <Form.Item
        label='Start date'
        name='startDate'
        rules={[{ required: true, message: 'Please input the start date!' }]}
      >
        <DatePicker data-testid='project-date-picker' />
      </Form.Item>
      <Form.Item
        label='Description'
        name='description'
      >
        <Input.TextArea rows={4} data-testid='project-description' />
      </Form.Item>

      <Form.Item {...tailLayout}>
        <Button type='primary' htmlType='submit' data-testid='create-save-button'>
          {project ? 'Save' : 'Create'}
        </Button>
      </Form.Item>
    </Form>
  )
}

ProjectForm.propTypes = {
  project: PropTypes.object,
  onFinish: PropTypes.func,
  onFinishFailed: PropTypes.func
}

export default ProjectForm
