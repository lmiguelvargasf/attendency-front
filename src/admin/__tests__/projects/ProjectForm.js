import React from 'react'
import { render, cleanup } from '@testing-library/react'
import ProjectForm from '../../projects/ProjectForm'
import moment from 'moment'

Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn()
  }))
})

describe('ProjectForm component', () => {
  afterEach(cleanup)

  describe('project is passed', () => {
    let component
    beforeEach(() => {
      const project = {
        title: 'This is a title',
        startData: moment('2020-04-28', 'YYYY-MM-DD'),
        description: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit'
      }
      component = (
        <ProjectForm project={project} onFinish={jest.fn()} onFinishFailed={jest.fn()} />
      )
    })
    it('matches snapshot', () => {
      const { asFragment } = render(component)
      expect(asFragment()).toMatchSnapshot()
    })

    it('displays button saying Save', () => {
      const { getByTestId } = render(component)
      expect(getByTestId('create-save-button')).toHaveTextContent(/^Save$/)
    })
  })

  describe('project is not passed', () => {
    let component

    beforeEach(() => {
      component = (
        <ProjectForm onFinish={jest.fn()} onFinishFailed={jest.fn()} />
      )
    })

    it('matches snapshot', () => {
      const { asFragment } = render(component)
      expect(asFragment()).toMatchSnapshot()
    })

    it('displays button saying Create', () => {
      const { getByTestId } = render(component)
      expect(getByTestId('create-save-button')).toHaveTextContent(/^Create$/)
    })
  })
})
