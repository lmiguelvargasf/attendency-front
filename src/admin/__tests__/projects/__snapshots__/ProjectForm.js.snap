// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectForm component project is not passed matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="ant-form ant-form-horizontal"
    data-testid="create-project-form"
    id="project"
  >
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-form-item-label"
      >
        <label
          class="ant-form-item-required"
          for="project_title"
          title="Title"
        >
          Title
        </label>
      </div>
      <div
        class="ant-col ant-col-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input"
              data-testid="project-title-input"
              id="project_title"
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-form-item-label"
      >
        <label
          class="ant-form-item-required"
          for="project_startDate"
          title="Start date"
        >
          Start date
        </label>
      </div>
      <div
        class="ant-col ant-col-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-picker"
            >
              <div
                class="ant-picker-input"
              >
                <input
                  data-testid="project-date-picker"
                  placeholder="Select date"
                  readonly=""
                  size="12"
                  title=""
                  value=""
                />
                <span
                  class="ant-picker-suffix"
                >
                  <span
                    aria-label="calendar"
                    class="anticon anticon-calendar"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="calendar"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-form-item-label"
      >
        <label
          class=""
          for="project_description"
          title="Description"
        >
          Description
        </label>
      </div>
      <div
        class="ant-col ant-col-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <textarea
              class="ant-input"
              data-testid="project-description"
              id="project_description"
              rows="4"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-col-offset-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <button
              class="ant-btn ant-btn-primary"
              data-testid="create-save-button"
              type="submit"
            >
              <span>
                Create
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;

exports[`ProjectForm component project is passed matches snapshot 1`] = `
<DocumentFragment>
  <form
    class="ant-form ant-form-horizontal"
    data-testid="create-project-form"
    id="project"
  >
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-form-item-label"
      >
        <label
          class="ant-form-item-required"
          for="project_title"
          title="Title"
        >
          Title
        </label>
      </div>
      <div
        class="ant-col ant-col-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <input
              class="ant-input"
              data-testid="project-title-input"
              id="project_title"
              type="text"
              value="This is a title"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-form-item-label"
      >
        <label
          class="ant-form-item-required"
          for="project_startDate"
          title="Start date"
        >
          Start date
        </label>
      </div>
      <div
        class="ant-col ant-col-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <div
              class="ant-picker"
            >
              <div
                class="ant-picker-input"
              >
                <input
                  data-testid="project-date-picker"
                  placeholder="Select date"
                  readonly=""
                  size="12"
                  title=""
                  value=""
                />
                <span
                  class="ant-picker-suffix"
                >
                  <span
                    aria-label="calendar"
                    class="anticon anticon-calendar"
                    role="img"
                  >
                    <svg
                      aria-hidden="true"
                      class=""
                      data-icon="calendar"
                      fill="currentColor"
                      focusable="false"
                      height="1em"
                      viewBox="64 64 896 896"
                      width="1em"
                    >
                      <path
                        d="M880 184H712v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H384v-64c0-4.4-3.6-8-8-8h-56c-4.4 0-8 3.6-8 8v64H144c-17.7 0-32 14.3-32 32v664c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V216c0-17.7-14.3-32-32-32zm-40 656H184V460h656v380zM184 392V256h128v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h256v48c0 4.4 3.6 8 8 8h56c4.4 0 8-3.6 8-8v-48h128v136H184z"
                      />
                    </svg>
                  </span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-form-item-label"
      >
        <label
          class=""
          for="project_description"
          title="Description"
        >
          Description
        </label>
      </div>
      <div
        class="ant-col ant-col-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <textarea
              class="ant-input"
              data-testid="project-description"
              id="project_description"
              rows="4"
            >
              Lorem ipsum dolor sit amet, consectetur adipiscing elit
            </textarea>
          </div>
        </div>
      </div>
    </div>
    <div
      class="ant-row ant-form-item"
    >
      <div
        class="ant-col ant-col-8 ant-col-offset-8 ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          >
            <button
              class="ant-btn ant-btn-primary"
              data-testid="create-save-button"
              type="submit"
            >
              <span>
                Save
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </form>
</DocumentFragment>
`;
